packages/dendron-plugin-views/src/components/DendronTreeExplorerPanel.tsx

```diff
@@ -15,12 +15,15 @@ type OnExpandFunc = TreeProps["onExpand"];
 type OnSelectFunc = TreeProps["onSelect"];
 const DendronTreeExplorerPanel: DendronComponent = (props) => {
   const logger = createLogger("DendronTreeExplorerPanel");
-  const noteActive = props.ide.noteActive;
   const engine = props.engine;
   const { config, notes } = engine;
+  const numNotes = _.size(notes);
+  const noteActive = props.ide.noteActive;
   const [activeNoteIds, setActiveNoteIds] = useState<string[]>([]);
   const [roots, setRoots] = useState<DataNode[]>([]);
-  const numNotes = _.size(notes);
+  // Used to avoid recomputing tree data unnecessarily
+  const [noteActiveId, setNoteActiveId] = useState<string>();
+  const [numNotesLast, setNumNotesLast] = useState<number>(numNotes);
```

Comment:
this would actually be useful for all views when calculating state. can you add this to [[../packages/common-frontend/src/features/ide/slice.ts]]?

introduce a `noteActivePrev` field and in [[setNoteActive|../packages/common-frontend/src/features/ide/slice.ts#L40]], set it to the active note. 
it will then get set automatically in [[../packages/dendron-plugin-views/src/components/DendronApp.tsx#L89]]